src/tracedAtomic.ml

```diff
@@ -54,13 +65,13 @@ let every_func = ref (fun () -> ())
 let final_func = ref (fun () -> ())
 
 (* Atomics implementation *)
-let atomics_counter = ref 1
+let atomics_counter = ref (-1)
 
 let make v = if !tracing then perform (Make v) else
     begin
       let i = !atomics_counter in
-      atomics_counter := !atomics_counter + 1;
-      (Atomic.make v, i)
+      atomics_counter := !atomics_counter - 1;
```

Comment:
Not added in this PR, but I think we should change `atomics_counter` to atomic - otherwise its a race outside tests.

Reply:
Yes absolutely, I'll switch the buggy ref to atomic!
(For context, I went with negative identifiers there to detect that a `trace` is interacting with an atomic created outside the tested function, but didn't follow up yet on that line of thoughts)